draft-ietf-moq-transport.md

```diff
@@ -1105,39 +1132,51 @@ to the old relay can be stopped with an UNSUBSCRIBE.
+associated with the set of tracks whose Track Namespace matches the namespace in
+an ANNOUNCE, or the Full Track Name in PUBLISH. Where the authorization and
+identification of the publisher occurs depends on the way the relay is managed
+and is application specific.
 
-A Relay can receive announcements from multiple publishers for the same
-Track Namespace and it SHOULD respond with the same response to each of the
-publishers, as though it was responding to an ANNOUNCE
-from a single publisher for a given track namespace.
+A Relay can receive announcements for the same Track Namespace or PUBLISH
+messages for the same track from multiple publishers, and it SHOULD respond with
+the same response to each of the publishers.
 
 When a publisher wants to stop new subscriptions for an announced namespace it
 sends an UNANNOUNCE. A subscriber indicates it will no longer subcribe to tracks
 in a namespace it previously responded ANNOUNCE_OK to by sending an
 ANNOUNCE_CANCEL.
 
-A relay manages sessions from multiple publishers and subscribers,
-connecting them based on the track namespace. This MUST use an exact
-match on track namespace unless otherwise negotiated by the application.
-For example, a SUBSCRIBE namespace=foobar message will be forwarded to
-the session that sent ANNOUNCE namespace=foobar.
-
-When a relay receives an incoming SUBSCRIBE request that triggers an
-upstream subscription, it SHOULD send a SUBSCRIBE request to each
-publisher that has announced the subscription's namespace, unless it
-already has an active subscription for the Objects requested by the
-incoming SUBSCRIBE request from all available publishers.
+A relay manages sessions from multiple publishers and subscribers, connecting
+them based on the track namespace or track name.  Prefix matching is used to
+determine which publishers receive a SUBSCRIBE or which subscribers receive a
+PUBLISH. For example, a SUBSCRIBE namespace=(foo,bar), track=x message will be
+forwarded to the sessions that sent ANNOUNCE namespace=(foo) and ANNOUNCE
+namespace=(foo, bar) respectively, but not one that sent ANNOUNCE
+namespace=(foobar).  Relays MUST forward to all publishers or subscribers that
+have a namespace prefix match.  TODO: do we need to say more about this.
+
+When a relay receives an incoming SUBSCRIBE request that triggers an upstream
+subscription, it SHOULD send a SUBSCRIBE request to each publisher that has
+announced the subscription's namespace or prefix thereof, unless it already has
+an active subscription for the Objects requested by the incoming SUBSCRIBE
+request from all available publishers.  If it already has a matching upstream
+subscription in Forward State=0, it SHOULD send a SUBSCRIBE_UDPATE with
+Forward=1 to all publishers.
+
+When a relay receives an incoming PUBLISH message, it SHOULD send a PUBLISH
+request to each subscriber that has subscribed to the track's namespace or
+prefix thereof.
```

Comment:
```suggestion
prefix or FullTrackName thereof.
```

Reply:
Should SUBSCRIBE(full track name) should solicit PUBLISH(full track name)?  The SUBSCRIBE will resolve by forwarding upstream.  If you are waiting for a particular track to show up (eg pending SUBSCRIBE), shouldn't you SUBSCRIBE_ANNOUNCES (namespace) ?